cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeHandler.java

```diff
@@ -317,12 +325,12 @@ public void chunk(ByteBuf request, HttpResponder responder) {
     public void finished(HttpResponder responder) {
       try {
         if (payloads.isEmpty()) {
-          responder.sendStatus(HttpResponseStatus.OK);
+          responder.sendJson(HttpResponseStatus.OK, GSON.toJson(runRecord, RunRecordDetail.class));
```

Comment:
Why are we sending back the whole run record? We should only send something back if the state is stopping? Also, it is better be a different data structure that carry less data than the whole run record detail.